core/src/test/java/org/infinispan/distribution/DistSyncFuncTest.java

```diff
@@ -306,27 +306,19 @@ public void testComputeFromNonOwner() throws InterruptedException {
       // compute function applied
       initAndTest();
       Object retval = getFirstNonOwner("k1").compute("k1", (k, v) -> "computed_" + k + "_" + v);
-      if (testRetVals) assertEquals("computed_k1_value", retval);
       asyncWait("k1", ComputeCommand.class);
-      assertOnAllCachesAndOwnership("k1", "computed_k1_value");
+      if (testRetVals) assertEquals("computed_k1_value", retval);
+      eventuallyAssertOnAllCachesAndOwneship("k1", "computed_k1_value");
```

Comment:
When I see a test that passes all the time for a while but in jenkins I get suddenly a => Expecting [k1] to equal [null] on cache [DistSyncFuncTest[DIST_SYNC, tx=false]-NodeB-50117] but was [computed_k1_value]. Owners are [Cache &apos;dist&apos;@DistSyncFuncTest[DIST_SYNC, tx=false]-NodeC-19559, Cache &apos;dist&apos;@DistSyncFuncTest[DIST_SYNC, tx=false]-NodeD-7512] 

telling me that a non owner has not deleted a value that was supposed to be deleted when looking directly in every cache, my guess is that the Command has not been executed yet as fast as it should be in every cache and the check is happening before that happens. The asyncWait of the command does nothing in the case of DistSyncFuncTest.

But you are right, the thing is that I'm not able to reproduce this problem. I'm going to open a ticket for this and check closer after I finish one thing I'm working on.